src/components/ContentLoader.tsx

```diff
@@ -0,0 +1,55 @@
+import { useEffect, useState } from 'react'
+import { ENV } from '../lib/env'
+
+interface ContentResponse {
+  id: string
+  content: Array<{variant: string, html: string}>
+}
+
+enum FetchStatus {
+  Unfetched,
+  FetchSuccess,
+  FetchFailure
+}
+
+export const ContentLoader = ({ contentId, contentLoadedCallback }: { contentId: string, contentLoadedCallback: () => void }): JSX.Element => {
+  const [innerHTMLContent, setInnerHTMLContent] = useState<string>('')
+  const [fetchStatus, setFetchStatus] = useState<Number>(FetchStatus.Unfetched)
+
+  const fetchContent = async (): Promise<void> => {
+    const request = new Request(`${ENV.OS_RAISE_CONTENT_URL_PREFIX}/${contentId}.json`)
+
+    try {
+      const response = await fetch(request)
+      const data = await response.json() as ContentResponse
```

Comment:
I'm confused about which things are running asynchronously here. Is it the case that multiple of these ContentLoader functional components are making their fetches simultaneously? If that is the case how are these two awaits working together? - do all the components wait on the fetches to finish, then for all the responses to be collected?